### PR TITLE
Logger dumps error with classname

### PR DESCRIPTION
**Pull request by tagomoris**
In these days, many plugin authors (and core developers) writes logging code like below many times.
```ruby
log.error "my message", error_class: e.class, error: e.to_s, other: attributes
```
Fairly speaking, this is waste of time and key typing.

Fluentd will log errors like below with this change by just specifying `error: e`:
```ruby
log.error "my message", error: e
# [error]: my message error=MyErrorClass[message of e]
```

### CODE REVIEW

**Comment by tagomoris** [comment]
@repeatedly @sonots How do you think about this change?

**Comment by sonots** [comment]
I am positive for specifying only an error object, but somebody might be watching logs of fluentd by a log monitoring system for alerting. Changing log format would break compatibilities for their system, so I think we should keep compatibilities of log format.

**Comment by tagomoris** [comment]
@sonots Do you mean that we should print errors in format below?
```
error_class=MyErrorClass error="message of e"
```

**Comment by sonots** [comment]
@tagomoris yes!

**Comment by tagomoris** [comment]
@sonots Hmm, it looks a bit verbose for me, and v0.14 may be a good time to do that kind of change.
@repeatedly How do you think?

**Comment by repeatedly** [comment]
Keeping compatibility is good for existing users. Maybe, asking it on mailing list is better.

> v0.14 may be a good time to do that kind of change.

This change affects v0.12 because we will backport several changes to v0.12 and skipping this patch is hard...
If you want to change error message format, we should suspend this patch until just before v0.14 release.

**Comment by tagomoris** [comment]
Okay, I'll change format of errors.

**Comment by tagomoris** [comment]
Done.

File: lib/fluent/log.rb [hunk 1]
```diff
@@ -292,7 +292,11 @@ def event(level, args)
       }
 
       map.each_pair {|k,v|
-        message << " #{k}=#{v.inspect}"
+        if k == "error".freeze
+          message << " error_class=#{v.class.to_s} error=#{v.to_s.inspect}"
```
Comment (sonots) [inline lib/fluent/log.rb]: `v` would be a string like https://github.com/fluent/fluentd/blob/88054a0b4299845c6696335876cba03bb33ea6e3/lib/fluent/output.rb#L127, so need to add `and v.is_a?(Exception)`.

Comment (sonots) [inline lib/fluent/log.rb]: Need to add another condition `!map.has_key?('error_class')`, otherwirse, ` error_class=` appears twice for old codes such as:

```
log.error 'message', error_class: e.class, error: e
```

Comment (tagomoris) [inline lib/fluent/log.rb]: That's right. I fixed so.

**Comment by sonots** [comment]
LGTM!

**Comment by tagomoris** [comment]
Merged.

**Pull request merged**